### PR TITLE
feature/KNDP-46:  Merged the build and releases workflow,  as 1 releases workflow with 1 deploy job

### PR DESCRIPTION
**Pull request by Enotmd**

### CODE REVIEW

File: .github/workflows/releases.yaml [hunk 1]
```diff
@@ -65,4 +72,25 @@ jobs:
...
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.ACCESS_TOKEN }}
 
+      - name: extract and build
+        id: build
+        run: |
+              git diff --name-only HEAD~1..HEAD > build.txt
+              export hash=$(echo "${{ github.sha }}")
+              cat build.txt | xargs -I {} dirname {} | awk -F'/' 'NF >= 2 { print $1 "/" $2 }' | uniq | xargs -I{} find {} -type f -name "Dockerfile" | xargs -I{} bash -c 'context=$(dirname "{}"); dir=$(dirname "{}" | awk -F"/" "{ print \$2 }"); docker build -t "${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}/$dir:$hash" "$context" && docker push "${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}/$dir:$hash"'
```
Comment (art10m) [inline .github/workflows/releases.yaml:96]: @Enotmd Please integrate this functionality into the deploy job instead of keeping it as a separate one.

**Review by art10m — CHANGES_REQUESTED** [review]

File: .github/workflows/releases.yaml [hunk 2]
```diff
@@ -65,4 +72,17 @@ jobs:
           github_token: ${{ secrets.ACCESS_TOKEN }}
           publish_dir: /home/runner/work/kndp/kndp/website/build
 
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.ACCESS_TOKEN }}
```
Comment (art10m) [inline .github/workflows/releases.yaml:80]: @Enotmd The step should be relocated to the top, where all the job configurations are performed.

File: .github/workflows/releases.yaml [hunk 3]
```diff
@@ -65,4 +72,17 @@ jobs:
...
 
+      - name: extract and build Docker images
+        id: build
+        run: |
+              cd /home/runner/work/kndp/kndp/
+              git diff --name-only HEAD~1..HEAD > build.txt
+              export hash=$(echo "${{ github.sha }}")
+              cat build.txt | xargs -I {} dirname {} | awk -F'/' 'NF >= 2 { print $1 "/" $2 }' | uniq | xargs -I{} find {} -type f -name "Dockerfile" | xargs -I{} bash -c 'context=$(dirname "{}"); dir=$(dirname "{}" | awk -F"/" "{ print \$2 }"); docker build -t "${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}/$dir:$hash" "$context" && docker push "${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}/$dir:$hash"'
```
Comment (art10m) [inline .github/workflows/releases.yaml:88]: @Enotmd Integrate this step above "Build website" step

**Review by art10m — CHANGES_REQUESTED** [review]

**Comment by Enotmd** [comment]
So, if you open the workflow from the action tab, its clearly building and pushing (on branch knpd46 at the moment), but it doesnt go to github packages. 
@art10m 
image

File: .github/workflows/releases.yaml [hunk 4]
```diff
@@ -4,23 +4,37 @@ on:
...
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.ACCESS_TOKEN }}   
+           
       - name: Checkout
         uses: actions/checkout@v1
 
+
```
Comment (art10m) [inline .github/workflows/releases.yaml:33]: @Enotmd remove this line

File: .github/workflows/releases.yaml [hunk 5]
```diff
@@ -37,10 +51,16 @@ jobs:
           CR_OWNER: web-seven
           CR_GIT_BASE_URL: "https://api.github.com/"
           CR_GIT_UPLOAD_URL: "https://uploads.github.com/"
-          CR_SKIP_EXISTING: false
+          CR_SKIP_EXISTING: true
```
Comment (art10m) [inline .github/workflows/releases.yaml:54]: @Enotmd Why did you switch it to true?

Comment (Enotmd) [reply .github/workflows/releases.yaml]: at one moment it didnt want to build cause it already existed and it failed the job, it was a problem of  docker registry login not in the right place. switched back

**Review by art10m — APPROVED** [review]

**Pull request merged**